Adds Avali + Vulp Noises

KillerTheFlareon opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
I propose to change three things.
1. Adding another species to starlight (cool space raptor bird thing)
2. Giving vulps unique chat noises, like slimes or others.
3. Updating vulp biology to have temperature stats like every other species.
UPDATE:
New chem called Amoxla added to PR, saline/dex/dex+ is also lethal to Avali now. (Amoxla takes the place of them, if you couldn't guess) and ammonia will heal airloss slightly.
Update 2: New sprites are in but are not shown in the videos below. 
## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
You know, I could give some overly complicated explanation but this is a matter of purely "I like this and I want to see this in-game" so, yeah. Plus people have been asking for avali for a while now, so I figured why not?

As for the vulps, it was rather odd that they didn't have ANY unique temperature values whatsoever, when even the base-game moth had it.

NOTE: I don't know surgery stuff so someone else will have to make them be able to be operated on (unless that's just a thing that applies to everything I have no clue.)
## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

https://github.com/user-attachments/assets/94410950-416e-4178-8075-b59a79f1d866

https://github.com/user-attachments/assets/4dc15251-3491-461f-b4be-e3f8b100f92c


https://github.com/user-attachments/assets/66cb990f-eed1-448a-8470-c647f8b94b91

A very sharply dressed boi showing off the nanite shell.

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: Killer Tamashi (Code), Zeyria (Sprites), Taro Varné (Unique Species features concepts), Fasuh (Avali Stasis Code)
- add: Added Avali (New playable race)
- add: Added Vulp speech noises/prefixes
- add: Added Vulp temperature ranges

KillerTheFlareon committed:
Adds Avali + Vulp Noises

KillerTheFlareon committed:
While i'm at it, fixes the plasma bug with .20 magazines.

KillerTheFlareon committed:
Avali and Vulp Race updates

KillerTheFlareon commented:
Yaml continues to be angry with me but it's not me specifically so yeee

KillerTheFlareon commented:
Closed on the request of a dev (which I agree with, to be honest) for a respite of the Avalis.

Rinary1 commented on Resources/Prototypes/_StarLight/Recipes/Lathes/security/ammo/rifle.yml:
```diff
@@ -61,7 +61,6 @@
   materials:
     Steel: 50
     Plastic: 100
-    Plasma: 250
```
????

KillerTheFlareon replied:
It's a bug that I fixed with the crafting recipe (or typo, however you wanna call it)

Rinary1 replied:
remove this

Rinary1 commented on Resources/ServerInfo/Guidebook/Mobs/Avali.xml:
```diff
@@ -0,0 +1,15 @@
+<Document>
+  # Avali
+
+  <Box>
+    <GuideEntityEmbed Entity="MobAvali" Caption=""/>
+  </Box>
+  Avali are a species of ammonia-based intelligent bird/raptor people and due to this odd combination many refer to them as "Space Raptors" Their unique biology allows them to deal exceedingly well with the cold, and is capable of basic flight in no to low gravity environments. 
+
+  ## Racial Features
+
+  - Receive [color=#ffa500]40% more fire damage[/color] and [color=#1e90ff]40% less cold damage[/color].
+  - Can stand in tempatures of up to -50c before taking cold damage
+  - Deal [color=red]5[/color] slashing damage with claws.
+  - Can fly in Zero-G.
+</Document>
```
it's just a moth copy

KillerTheFlareon replied:
That's how avali are designed in the place it originates from (shrug)
As for why, they look different and are already in other MRP servers, so all it really adds is variety to species selection. (Like how there's dwarfs, which are really just a copy of humans and not much else.)

The PR's closed right now so if you wanna suggest something unique I'm all ears.

KillerTheFlareon replied:
Could hear one tile further? They do have 4 ears, and big ones at that.

Rinary1 reviewed (commented):
it's just a moth copy, why we need to add this?

KillerTheFlareon committed:
Avali update + Chem

A new chem made specifically for avali to treat airloss/bloodloss, acts as poison in other non-avali

Dex, dex plus and saline is highly toxic to them, as well as handful of changes in the species

KillerTheFlareon commented:
Major update, will edit page to reflect new features.
Resprite isn't done yet so it's staying closed for now.
(I think it only updates when it's open?)

KillerTheFlareon committed:
Resprite!

KillerTheFlareon committed:
More updoots

KillerTheFlareon commented:
the sprites are here, the sounds are nice (if not a bit placeholder-y)
it is time.

KillerTheFlareon commented:
wait a second..where's my chem stuff- one second

KillerTheFlareon commented:
they're there in the code. Just not showing up in update history, weirdly enough.

KillerTheFlareon committed:
Typo fixed

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Mobs/Species/avali.yml:
```diff
@@ -0,0 +1,147 @@
+- type: entity
+  save: false
+  name: Urist McRaptor
```
Rename(McRaptor it's resomi)

Rinary1 commented on Resources/Prototypes/_StarLight/Entities/Mobs/Species/avali.yml:
```diff
@@ -0,0 +1,147 @@
+  - type: Speech
+    speechVerb: Parrot # RMC did it right.
+    speechSounds: MaleAvali
+  - type: PassiveDamage 
+    allowedStates:
+    - Alive
+    damageCap: 180 #To clarify due to the lack of cap, if you're knocked into crit you still won't be able to get back up even with passive healing. Weaker than normal regen, too.
+    damage:
+      types:
+        Heat: -0.05
+      groups:
+        Brute: -0.05
```
remove passive heal

Rinary1 commented on Resources/Prototypes/Reagents/botany.yml:
```diff
@@ -214,9 +214,23 @@
     Poison:
       effects:
       - !type:HealthChange
+        conditions: #Staarliight
```
Starlight

Rinary1 commented on Resources/Prototypes/Reagents/botany.yml:
```diff
@@ -214,9 +214,23 @@
         damage:
           types:
             Caustic: 1
+    Medicine: #Starliiiight
```
Starlight

Rinary1 commented on Resources/Prototypes/Reagents/elements.yml:
```diff
@@ -69,6 +69,14 @@
+          shouldHave: false
+        damage:
+          types:
+            Poison: 0.2
```
more poison damage(it steel useful if you use dylo)

Rinary1 commented on Resources/Prototypes/Reagents/elements.yml:
```diff
@@ -160,6 +168,14 @@
+          shouldHave: true
+        damage:
+          types:
+            Poison: 0.2
```
more poison damage(it steel useful if you use dylo)

KillerTheFlareon commented:
Changes applied but not pushed, waiting on passive heal judgement. Will close for now due to.. a gross misinterpretation on the actual values the passive healing is..well, healing.
Whoever said that 0.7 blunt/burn heals 4 damage per minute lied. And due to that lie, I rightfully assumed 0.5 would heal around ~2 damage per minute. It does not even come CLOSE.
And I based all my stuff around it.. so yeah, gonna need to look over the values again because they are hilariously low, pending talks with the devs about this feature in the first place.

KillerTheFlareon committed:
Overhaul v1

Ugh.. C#

KillerTheFlareon commented:
On hiatus until I can find a C# coder to code in something

KillerTheFlareon commented:
Coder found, will try to put this pr in draft mode..if i figure out how to

PubliclyExecutedPig commented:
is this still being worked on?

KillerTheFlareon commented:
> is this still being worked on?

It's soon(TM)
The C# coder is almost done and then I need to unfuck the pr with the massive amounts of merge conflicts this has gained for being in dev for so long

bartlomiejszm committed:
Add Avali Stasis Action

KillerTheFlareon committed:
Merge pull request #1 from Fasuh/vali-tiem-stasis

Add Avali Stasis Action

KillerTheFlareon commented:
Too complex to resolve in editor?? goddamnit

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
Tada!

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon commented:
Burnout be dammed, I'm getting this pr out the door as fast as I can

KillerTheFlareon committed:
Changes

KillerTheFlareon committed:
changes

KillerTheFlareon committed:
Links in Avali to The Nexus

:D

bartlomiejszm committed:
Fixes, and added animations.

KillerTheFlareon committed:
awa

KillerTheFlareon committed:
awa

bartlomiejszm committed:
Fix crash on resistance check

KillerTheFlareon committed:
textures

KillerTheFlareon committed:
Merge branch 'pr/2' into vali-tiem

KillerTheFlareon committed:
spritening

bartlomiejszm committed:
Added Animation on exiting stasis, and an animation for being in stasis, additionally added bleed healing properties

KillerTheFlareon committed:
Update avali.yml

KillerTheFlareon committed:
Merge branch 'pr/2' into vali-tiem

KillerTheFlareon committed:
Nanomachines, son

bartlomiejszm committed:
Update avali stasis animations to be correctly synchronized with other clients

bartlomiejszm committed:
Updated avali stasis animations to be correctly synchronized with other clients

bartlomiejszm committed:
Merge branch 'vali-tiem' into vali-tiem-stasis

KillerTheFlareon committed:
Merge pull request #5 from Fasuh/vali-tiem-stasis

Vali tiem stasis

KillerTheFlareon committed:
wawa

KillerTheFlareon committed:
updates

bartlomiejszm committed:
feat(AvaliStasis): Added logic to lower the healing rate when user is in crit status, and remove stasis when they die

bartlomiejszm committed:
Updated animation to not rotate when user moves, additionally delayed the activation of stasis, to match the animation.

KillerTheFlareon committed:
Merge pull request #6 from Fasuh/vali-tiem-stasis

Vali tiem stasis

bartlomiejszm committed:
fix(AvaliStasis): Delay continuous animation to remove a perceived flicker

KillerTheFlareon committed:
funny ourple blood

KillerTheFlareon committed:
Merge branch 'pr/7' into vali-tiem

KillerTheFlareon commented:
YAML ERROR NOOO

KillerTheFlareon committed:
bugfix

bartlomiejszm committed:
chore(AvaliStasis): Fix linter analysis and add bleeding in critical status value

KillerTheFlareon committed:
Merge branch 'ss14Starlight:Starlight' into vali-tiem

bartlomiejszm committed:
fix(AvaliStasis): fix issue where when user exited stasis while in preparing animation, it could mess up the animations.

Fasuh committed:
Merge branch 'vali-tiem' into vali-tiem-stasis

KillerTheFlareon commented:
god

KillerTheFlareon committed:
Update avali.yml

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

Happyrobot33 commented:
Gonna be commenting things here as i think about them

Surgery needs to be checked/implemented

They need a species tag assigned to them like resomi or felionoids

They need knockback setup for them, somewhere between resomi and felionoids values

Resomi need to inherit the new biology (i can help with this)

KillerTheFlareon commented:
> Gonna be commenting things here as i think about them
> 
> Surgery needs to be checked/implemented
> 
> They need a species tag assigned to them like resomi or felionoids
> 
> They need knockback setup for them, somewhere between resomi and felionoids values
> 
> Resomi need to inherit the new biology (i can help with this)


- Surgery works
- Already have a tag
- Any suggestions for a number? They're more sturdy and a bit bigger than a resonmi.
- (I would appreciate you doing this part, yeah.)

bartlomiejszm committed:
Merge remote-tracking branch 'starlight/Starlight' into vali-tiem-stasis

# Conflicts:
#	Resources/Audio/_Starlight/Effects/attributions.yml
#	Resources/Prototypes/_StarLight/Entities/Mobs/Species/vulpkanin.yml
#	Resources/Prototypes/_StarLight/Reagents/biological.yml

bartlomiejszm committed:
Merge remote-tracking branch 'starlight/Starlight' into vali-tiem-stasis

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
awa

KillerTheFlareon committed:
wawa

bartlomiejszm committed:
Merge remote-tracking branch 'killer/vali-tiem' into vali-tiem-stasis

# Conflicts:
#	Resources/Audio/_Starlight/Effects/attributions.yml

KillerTheFlareon committed:
trying to fix this shit

bartlomiejszm committed:
chore(AvaliStasis): Add comments to code

KillerTheFlareon committed:
Merge pull request #9 from Fasuh/vali-tiem-stasis

Vali tiem stasis

Happyrobot33 commented on Content.Client/_Starlight/Avali/Systems/ClientAvaliStasisSystem.cs:
this folder path feels too specifc. Try to follow the resomi jump skill one (well ofc we redid that one but what it was before)

Happyrobot33 commented on Resources/Prototypes/_StarLight/Entities/Mobs/Player/avali.yml:
```diff
@@ -0,0 +1,7 @@
+- type: entity
+  save: false
+  name: Urist McRaptor
```
needs a different name, this will be confusing in the spawn menu

KillerTheFlareon replied:
Changed

Happyrobot33 commented on Resources/Textures/_Starlight/Effects/avali_nanites.rsi/nanite-building.png:
is there a  reason these two are seperate building and unbuilding sprites instead of just playing it backwards?

KillerTheFlareon replied:
Didn't work when I tried doing that, hence them being separate

Happyrobot33 commented on Content.Client/_Starlight/Avali/Systems/ClientAvaliStasisSystem.cs:
```diff
@@ -0,0 +1,160 @@
+/// <summary>
+/// Client-side system that handles visual and audio effects for Avali stasis.
+/// </summary>
+public sealed class ClientAvaliStasisSystem : SharedAvaliStasisSystem
```
check the naming scheme of this, I dont think Client is normally put for systems inheriting from shared?

Happyrobot33 commented on Content.Client/_Starlight/Avali/Systems/ClientAvaliStasisSystem.cs:
```diff
@@ -0,0 +1,160 @@
+using Robust.Client.GameObjects;
+using DrawDepth = Content.Shared.DrawDepth.DrawDepth;
+
+namespace Content.Client.Starlight.Avali.Systems;
```
this namespace feels too specific. Check resomi jump skill namespacing

Happyrobot33 reviewed (changes requested):
Overall looks fairly good. Still need to make the changes to resomi, but I might actually end up doing those myself in a seperate PR after this one merges to keep things simpler

Happyrobot33 committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
Update avali.yml

tada

KillerTheFlareon committed:
Some light fixes

Happyrobot33 commented on Resources/Prototypes/_StarLight/Species/avali.yml:
```diff
@@ -0,0 +1,192 @@
+  name: Avali Stasis
+  description: Allows Avali to enter nanite induced stasis.
+  components:
+    - type: AvaliStasis
```
imo all of these values should be the "default" state of the component so we dont have this wall of text

KillerTheFlareon replied:
It's like this so it can be easily modified, if someone wants to use this code in the future.

Happyrobot33 replied:
someone can modify the comp later by adding the fields they want to edit

KillerTheFlareon replied:
Aye

Happyrobot33 commented on Resources/Prototypes/_StarLight/Entities/Mobs/Species/avali.yml:
```diff
@@ -0,0 +1,196 @@
+    shiveringHeatRegulation: 2000
+    normalBodyTemperature: 261
+    thermalRegulationTemperatureThreshold: 1
+  - type: AvaliStasis
```
similar to the other review, make these defaults

KillerTheFlareon replied:
Can't use the default values, it had to be modified to allow their body to exist at -17c instead of the normal 35(iirc?)c

Happyrobot33 replied:
im specifically talking about the stasis component

KillerTheFlareon replied:
ah, that is a job for faush then

Happyrobot33 commented on Resources/Locale/en-US/metabolism/metabolizer-types.ftl:
starlight comments

Happyrobot33 commented on Resources/Locale/en-US/reagents/meta/botany.ftl:
starlight edit comment

KillerTheFlareon replied:
How? the normal way to comment (#) doesn't seem to work in ftl files

Happyrobot33 commented on Resources/Locale/en-US/reagents/meta/medicine.ftl:
starlight comment

Happyrobot33 commented on Resources/Prototypes/Reagents/elements.yml:
starlight comment

KillerTheFlareon replied:
Already did? It's right there

Happyrobot33 reviewed (changes requested):
another review, im up to 160/181 files. I need to test all the audio for the vibes

KillerTheFlareon commented:
Suffering rn

KillerTheFlareon committed:
Adler is a gigachad

Heal action fixed

KillerTheFlareon committed:
Merge branch 'Starlight' into vali-tiem

KillerTheFlareon committed:
Fixes

KillerTheFlareon committed:
Starlight line changes

KillerTheFlareon committed:
awa

KillerTheFlareon committed:
Knockback added

bartlomiejszm committed:
chore(Avali): Fixed issues from PR comment regarding naming and default values

Happyrobot33 commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:
```diff
@@ -15,6 +15,9 @@
         knockback: 10
         staminaMultiplier: 8
+      Avali:
+        knockback: 3 # Bigger bird, more sturdy bones.. or something.
```
avali are less suceptive to knockback than the cats?

KillerTheFlareon replied:
Yeah. Think about it, the cats are small as hell and a shotgun blast would effect them pretty severely. Resonmi are a bit smaller than Avali but much more frail

Avali by comparison are the biggest of the two, built a bit more sturdy, so they could reasonably handle recoil better.

So if resonmi is weakest, felind is second weakest, Avali would be third

KillerTheFlareon replied:
And yes I know where I commented, it was an example that applies to the rest

Happyrobot33 commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:
```diff
@@ -15,6 +15,9 @@
         knockback: 3
         staminaMultiplier: 20
+      Avali:
+        knockback: 1 # Bigger bird, more sturdy bones.. or something.
```
less than felionoids?

Happyrobot33 commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:
```diff
@@ -15,6 +15,9 @@
         knockback: 4
         staminaMultiplier: 15
+      Avali:
+        knockback: 1 # Bigger bird, more sturdy bones.. or something.
```
less than felionoids?

Happyrobot33 commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:
need to update the RPG entry

KillerTheFlareon replied:
Done

Happyrobot33 reviewed (changes requested):
im not sure if avali should be considered more lightweight than the cats

bartlomiejszm committed:
chore(Avali): Fix linter errors

bartlomiejszm committed:
chore(Avali): Yet another linter fixes

bartlomiejszm committed:
chore(Avali): Hopefully last fix after linter errors

KillerTheFlareon committed:
Last fix hopefully

KillerTheFlareon commented:
Happy. PLEASE.

Happyrobot33 committed:
Merge branch 'Starlight' into vali-tiem
